src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -948,25 +949,52 @@ RangesInDataParts MergeTreeDataSelectExecutor::filterPartsByPrimaryKeyAndSkipInd
 
     std::list<DataSkippingIndexAndCondition> useful_indices;
     std::map<std::pair<String, size_t>, MergedDataSkippingIndexAndCondition> merged_indices;
+    std::unordered_set<std::string> ignored_index_names;
+
+    if (use_skip_indexes && settings.ignore_data_skipping_indices.changed)
+    {
+        const auto & indices = settings.ignore_data_skipping_indices.toString();
+        Tokens tokens(indices.data(), &indices[indices.size()], settings.max_query_size);
```

Comment:
It's better to use `indices.data() + indices.size()`, since `indices[indices.size()]` is an UB, I think.